File: code-dist-2019-10/MATLAB/rei4p18/min_distance.m
mat_data = [2 4 5 1 3; 4 2 3 3 1];
d = size(mat_data,1);  r = size(mat_data,2);
cvx_begin
    variable v(d,1)
    variable z(r,1)
    minimize( sum(z) )
    subject to
        for l=1:r
            z(l) >= norm(v - mat_data(:,l));
        end
cvx_end

File: code-dist-2019-10/Python/rei4p18/min_distance.py
import cvxpy as cp
import numpy as np
S = np.array([
    [2.0, 4.0, 5.0, 1.0, 3.0],
    [4.0, 2.0, 3.0, 3.0, 1.0] ])
d, r = S.shape[0], S.shape[1]
v, z = cp.Variable(d), cp.Variable(r)
obj  = cp.Minimize( sum(z) )
cons = []
for l in range(0,r):
    cons += [
        z[l] >= cp.norm(v - S[:,l])
    ]
P = cp.Problem(obj, cons)
P.solve(verbose=True)
print(v.value)
